File: src/test/java/homework7/BaseTestAmazon.java
package homework7;

import BaseTest.BaseTest;
import org.openqa.selenium.By;
import org.openqa.selenium.NoSuchElementException;
import org.openqa.selenium.TimeoutException;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;


class BaseTestAmazon extends BaseTest {
    final static String BASEURL = "https://www.amazon.com/";

    protected void choiceField(String fieldName) {
        try {
            WebElement s = new WebDriverWait(driver, Duration.ofSeconds(3)).until(ExpectedConditions.presenceOfElementLocated(By.xpath("//header")));
        } catch (TimeoutException e) {
            throw new NoSuchElementException(e.toString());
        }

        WebElement dropdownMenu = driver.findElement(By.xpath("//select[@id='searchDropdownBox']"));
        Select select = new Select(dropdownMenu);
        select.selectByVisibleText(fieldName);
    }
}

File: homework7/BaseTestAmazon.py
import pytest
from selenium import webdriver
from selenium.common import TimeoutException, NoSuchElementException
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as ec


@pytest.mark.usefixtures("chrome_driver")
class BaseTestAmazon:
    driver: webdriver.Chrome
    BASE_URL = "https://www.amazon.com"

    def choice_field(self, field_name):
        try:
            WebDriverWait(self.driver, 15).until(ec.element_to_be_clickable(self.driver.find_element(By.XPATH, "//header")))
        except TimeoutException as e:
            raise NoSuchElementException(e.msg)

        drop_down_menu = self.driver.find_element(By.XPATH, "//select[@id='searchDropdownBox']")
        select_field = Select(drop_down_menu)
        select_field.select_by_visible_text(field_name)
